src/binding.cc

```diff
@@ -684,81 +669,80 @@ class Client : public ObjectWrap {
           cur_query.column_count = n_fields;
           for (f = 0; f < n_fields; ++f) {
             field = fields[f];
-            cur_query.column_names[f] = Persistent<String>::New(
-                                            String::New(field.name,
-                                                        field.name_length)
+            NanAssignPersistent(cur_query.column_names[f],
+                                            NanNew<String>(field.name, field.name_length)
                                         );
           }
         }
-        row = Object::New();
+        row = NanNew<Object>();
         if (config.metadata) {
-          types = Object::New();
-          charsetNrs = Object::New();
-          dbs = Object::New();
-          tables = Object::New();
-          orgTables = Object::New();
-          names = Object::New();
-          orgNames = Object::New();
+          types = NanNew<Object>();
+          charsetNrs = NanNew<Object>();
+          dbs = NanNew<Object>();
+          tables = NanNew<Object>();
+          orgTables = NanNew<Object>();
+          names = NanNew<Object>();
+          orgNames = NanNew<Object>();
         }
       }
       for (f = 0; f < n_fields; ++f) {
         if (cur_query.row[f] == NULL)
-          field_value = Null();
+          field_value = NanNull();
         else if (IS_BINARY(fields[f])) {
           vlen = lengths[f];
           buf = (unsigned char*)(cur_query.row[f]);
           new_buf = new uint16_t[vlen];
           for (i = 0; i < vlen; ++i)
             new_buf[i] = buf[i];
-          field_value = String::New(new_buf, vlen);
+          field_value = NanNew<String>(new_buf, vlen);
           delete[] new_buf;
         } else
-          field_value = String::New(cur_query.row[f], lengths[f]);
+          field_value = NanNew<String>(cur_query.row[f], lengths[f]);
 
         if (cur_query.use_array)
           row->Set(f, field_value);
         else
-          row->Set(cur_query.column_names[f], field_value);
+          row->Set(NanNew<String>(cur_query.column_names[f]), field_value);
```

Comment:
I think that `NanNew` is unnecessary here since `column_names` already contains strings created with `NanNew` (on line 672-673 of your version).